File: solutions/_1608_special_array_with_x_elements_greater_than_or_equal_x/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var specialArray = function (nums) {
  nums.sort((a, b) => a - b);
  let prev = -Infinity;
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    const remaining = nums.length - i;
    if (num >= remaining && remaining > prev) {
      return remaining;
    }
    prev = num;
  }
  return -1;
};

module.exports = { specialArray };

/*
https://leetcode.com/problems/special-array-with-x-elements-greater-than-or-equal-x/
*/

File: solutions/_1608_special_array_with_x_elements_greater_than_or_equal_x/solution.py
from typing import List


class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort()
        prev = -float("inf")
        for i, num in enumerate(nums):
            remaining = len(nums) - i
            if num >= remaining > prev:
                return remaining
            prev = num
        return -1


# https://leetcode.com/problems/special-array-with-x-elements-greater-than-or-equal-x/
